qutebrowser/mainwindow/mainwindow.py

```diff
@@ -483,9 +487,14 @@ def _connect_signals(self):
     @pyqtSlot(bool)
     def _on_fullscreen_requested(self, on):
         if on:
+            self._restore_fullscreen = self.isFullScreen()
             self.showFullScreen()
-        else:
+        elif not self._restore_fullscreen:
             self.showNormal()
+        else:
+            self._restore_fullscreen = self.isFullScreen()
```

Comment:
I don't understand this line. When the window was fullscreen when a video was fullscreened, and the video now goes out of fullscreen, we should just do nothing, no? Why store the fullscreen state again?